File: LeetCode/Greedy/MinimizeMaximumPairSumInArray.java
// https://leetcode.com/problems/minimize-maximum-pair-sum-in-array/?envType=problem-list-v2&envId=greedy

class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int max = 0 , l =0 , r = nums.length-1;
        while (l < r) {
            int sum = nums[l] + nums[r];
            max = Math.max(max, sum);
            l++;
            r--;
        }
        return max;
    }
}

File: LeetCode/Two Pointers/MinimizeMaximumPairSuminArray.cs
///  https://leetcode.com/problems/minimize-maximum-pair-sum-in-array/description/


public class Solution {
    public int MinPairSum(int[] nums) {
        Array.Sort(nums);
        int left = 0, right = nums.Length-1, ans = 0;

        while(left < right)
            ans = Math.Max(ans, nums[left++] + nums[right--]);

        return ans;    
    }
}